diff --git a/drivers/gpu/arm/midgard/mali_kbase_gpu_memory_debugfs.c b/drivers/gpu/arm/midgard/mali_kbase_gpu_memory_debugfs.c
index e2948b1..23c36a4 100755
--- a/drivers/gpu/arm/midgard/mali_kbase_gpu_memory_debugfs.c
+++ b/drivers/gpu/arm/midgard/mali_kbase_gpu_memory_debugfs.c
@@ -48,10 +48,13 @@ static int kbasep_gpu_memory_seq_show(struct seq_file *sfile, void *data)
 		list_for_each_entry(element, &kbdev->kctx_list, link) {
 			/* output the memory usage and cap for each kctx
 			* opened on this device */
-			ret = seq_printf(sfile, "  %s-0x%p %10u\n", \
-				"kctx",
+			ret = seq_printf(sfile, "  %s-0x%p %10u %10u %10u %10u\n", \
+				"kctx", \
 				element->kctx, \
-				atomic_read(&(element->kctx->used_pages)));
+				element->kctx->pid, \
+				element->kctx->osalloc.free_list_max_size, \
+				atomic_read(&(element->kctx->used_pages)), \
+				atomic_read(&(element->kctx->nonmapped_pages)));
 		}
 		mutex_unlock(&kbdev->kctx_list_lock);
 	}
